### PR TITLE
[CPU] Coverity fixes

### PR DESCRIPTION
**Pull request by egorshul**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/src/utils/jit_kernel.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ void freeReg(jit_kernel::reg_indices & freeRegs, const registers<RegType> & regs
     //     throw std::runtime_error("Some register was freed twice");
     freeRegs.emplace_back(idx);
     if (freeRegs.size() > regs.size())
-        IE_THROW() << "Some register was freed twice";
+        throw std::runtime_error("Some register was freed twice");
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/utils/jit_kernel.cpp:36]: I guess it was changed on IE_THROW in the previous PR and now it is reverted. Why? What exactly coverity "says"?

File: src/plugins/intel_cpu/src/utils/jit_kernel.hpp [hunk 2]
```diff
@@ -668,8 +668,12 @@ variable_base<T>::variable_base(variable_base && rhs)
 
 template<typename T>
 variable_base<T>::~variable_base() {
-    if (_manage_lifetime)
-        kernel.free(_reg);
+    try {
+        if (_manage_lifetime)
+            kernel.free(_reg);
+    } catch (std::exception &e) {
+        std::cerr << e.what() << std::endl;
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/utils/jit_kernel.hpp:675]: Not sure this is good behavior. What's the coverity error?

Comment (egorshul) [reply src/plugins/intel_cpu/src/utils/jit_kernel.hpp:675]: In MKLDNNPlugin::​internal::​variable_base<unsigned long const>::​~variable_base(): A C++ exception is thrown but never caught. We must catch the error in the destructor.

Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/utils/jit_kernel.hpp:675]: Still think just printing error message in the stream is not a good way of handling. @vladislav-volkov Maybe you can suggest smt.

Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/utils/jit_kernel.hpp:675]: This issue should be fixed on master. Could you please rebase to latest version?
Destructors should never throw exceptions. So we can only catch them and suppress.

**Comment by egorshul** [comment]
This issue is fixed on the current master.

**Pull request closed without merging**